hazelcast-hibernate/hazelcast-hibernate4/src/test/java/com/hazelcast/hibernate/HibernateStatisticsTestSupport.java

```diff
@@ -205,4 +221,38 @@ public void testUpdateQueryCausesInvalidationOfEntireRegion() {
         assertEquals(10, dummyEntityCacheStats.getMissCount());
         assertEquals(0, dummyEntityCacheStats.getHitCount());
     }
+
+    @Test
+    public void testNaturalIdCacheStillHitsAfterIrrelevantNaturalIdUpdate() {
```

Comment:
I would do the test as below. 
insert(1)
query inserted entity.
assertCacheHit
update inserted entity.
assertMissHit

What do you think? That would be simpler.